### DISCUSSION TITLE
Custom SignaureValidator error: not a valid signature validator class.

### DISCUSSION THREAD

**Discussion by Mozzarella-dev** [spatie/laravel-webhook-client#124] [Q&A]
I really don't know if it's a bug or i'm missing something here.

This is my validator, its path is `App\Handler\MyValidator.php` :
```php
use Illuminate\Http\Request;
use Spatie\WebhookClient\WebhookConfig;
use Spatie\WebhookClient\Exceptions\InvalidConfig;
use Spatie\WebhookClient\SignatureValidator\SignatureValidator;

class MyValidator implements SignatureValidator
{
    public function isValid(Request $request, WebhookConfig $config): bool
    {
        $signature = $request->header($config->signatureHeaderName);

        if (! $signature) {
            return false;
        }

        $signingSecret = $config->signingSecret;

        if (empty($signingSecret)) {
            throw InvalidConfig::signingSecretNotSet();
        }

        $computedSignature = base64_encode(hash_hmac('sha256', $request->getContent(), $signingSecret, true));

        return hash_equals($signature, $computedSignature);
    }
}
```

In the webhook-client.php config file i've got:
```php
'signature_validator' => \App\Handler\MyValidator::class,
```

The full error message is the following:
```log
local.ERROR: `App\Handler\MyValidator` is not a valid signature validator class. A valid signature validator is a class that implements `Spatie\WebhookClient\SignatureValidator\SignatureValidator`. {"exception":"[object] (Spatie\\WebhookClient\\Exceptions\\InvalidConfig(code: 0): `App\\Handler\\MyValidator` is not a valid signature validator class. A valid signature validator is a class that implements `Spatie\\WebhookClient\\SignatureValidator\\SignatureValidator`. at /var/www/html/vendor/spatie/laravel-webhook-client/src/Exceptions/InvalidConfig.php:22)
```

**Comment by Mozzarella-dev** [comment]
Still didn't found a way around that error, if anybody could help would be great, thanks

**Reply by ryanmortier** [reply]
You're missing the namespace on your MyValidator class?

**Reply by Mozzarella-dev** [reply]
I don't think the namespace is the problem as long as i import everything correctly

**Reply by bartclaeys** [reply]
Did you even figure it out? I am dealing with the exact same error and my code is very similar to yours

**Reply by Mozzarella-dev** [reply]
> Did you even figure it out? I am dealing with the exact same error and my code is very similar to yours

At the end I went and modified directly the `Signature Validator` inside the `Vendor` folder.

Not elegant and need to remember that when updating, but works

**Reply by bartclaeys** [reply]
Thanks for the reply, that's very hacky :(

**Reply by alberto-dumontt** [reply]
Qual alteração que fez diretamente na Signature Validator???

**Comment by dsneath** [comment]
Experiencing a similar problem:


```
Custom Validator (TestingSignatureValidator.php)
<?php
namespace App\Handler;

use Illuminate\Http\Request;
use Spatie\WebhookClient\WebhookConfig;
use Spatie\WebhookClient\Exceptions\InvalidConfig;
use Spatie\WebhookClient\SignatureValidator\SignatureValidator;

class TestingSignatureValidator implements SignatureValidator
{
    public function isValid(Request $request, WebhookConfig $config): bool
    {
        return true;
    }
}
```


webhook-client.php

`'signature_validator' => \App\Handler\TestingSignatureValidator::class,`

Error via Postman:
`Spatie\WebhookClient\Exceptions\InvalidConfig: `App\Handler\TestingSignatureValidator` is not a valid signature validator class. A valid signature validator is a class that implements `Spatie\WebhookClient\SignatureValidator\SignatureValidator`. in file /var/www/laravel/vendor/spatie/laravel-webhook-client/src/Exceptions/InvalidConfig.php on line 22`

**Comment by Mozzarella-dev** [comment]
Can we tag this as an issue? I don't think at this point we're doing something wrong.
@freekmurze

**Reply by freekmurze** [reply]
Could you PR a failing test (and if you know the fix, a fix)?

**Reply by bartclaeys** [reply]
Confused about the ask for a PR for a test... Postman shows the above error, isn't that a test?

**Comment by dsneath** [comment]
I resolved mine myself, it was a rather embarrassing omission. My handler was in \App\Http\Handlers namespace, not \App\Handlers. Found by commenting out the is_subclass_of check on line 41 of src\WebhookConfig.php which then threw a "class does not exist" error; decided it had to be something wrong with my namespacing

@Mozzarella-dev obviously your issue may be different but I would check; it didn't help that I followed a tutorial that was based on Laravel 7.x and I'm using 9.x which changed how namespaces work

**Reply by Scrivs** [reply]
What's the change in how namespaces work?

**Comment by nuevocosas** [comment]
I had the same issue for a couple of hours today. Tried a bunch of debugging and troubleshooting but .... the issue was neither namespaces, classes, code errors or whatever. It was my file name.... Instead of CustomSignatureValidator.php it was just CustomSignatureValidator .

Best part was everything ran as per normal as if it is a PHP file, but just that the extension was not in..
